### ISSUE TITLE
Improve the Contrast Accessibility Warning

### ISSUE THREAD

**Issue by MadtownLems** [WordPress/gutenberg#34968]
## What problem does this address?

The Block Editor provides a great warning when background/text colors lack sufficient contrast to be easily readable, but could perhaps be improved to specifically call it out as an "Accessibility"-related issue.  We have been educating and pressing our content providers hard to improve the accessibility of their content, and I think that using that word here specifically would help people understand the gravity of the situation and make them take it more seriously.

Image Description: 
![Image of the WordPress contrast warning - This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color. ](https://user-images.githubusercontent.com/945157/134044574-ae87eeca-e3da-48c6-a68c-d22448bfd74d.png)

## What is your proposed solution?

Specifically cite that this is an "Accessibility" concern in the notification.
The current text of the warning is: "This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color."

There are lots of options for a solution. I'll outline a couple, but this ticket is being created at the request of the #accessibility chat where they wanted additional feedback or suggestions.

a) Add some text to the end, such as: "..... or a darker text color to make the content more accessible."
b) Add a helpful link below the paragraph, worded something like: "Learn More About Contrast Accessibility" that links to a trusted resource
c) Add a small heading before the warning, saying something like: "Accessibility Warning" or - to be less scary - something more like "This content could be more accessible."

**Comment by annezazu** [comment]
Thanks so much for opening this up! This seems like a great improvement. What would you like to see for a trusted resource? I think it would be great to link to one.

**Comment by MadtownLems** [comment]
>  What would you like to see for a trusted resource?

I'm not an expert, but this looks good to me
https://webaim.org/articles/contrast/

**Comment by javierarce** [comment]
Here are some ideas for the design of the messages. Besides adding a link and using the words "accessible" or "accessibility", I improved the message so it's easier to read and more actionable:

``` 
This color combination may be hard to read. Make it more accessible by using: 
- A darker text color. 
- A brighter background color. 
```


In the first row I show: 

1) A message with a link integrated into the text (so the message is not very long, but also points to more information)
2) Same but with an indication of the "problematic" color combination.
3) An idea to show more specific information (because of that, maybe this kind of info should be behind a setting)

The second row contains the same information, but the link appears at the bottom. I personally prefer the options of the first row, because the link is integrated and the box is not very tall.

![The image shows six designs for the color notification. The description for each of them appears in the body of the issue](https://user-images.githubusercontent.com/4933/134165162-2529826d-d05c-4cc3-853f-8ad9e11aae69.png)

**Comment by karmatosed** [comment]
@javierarce some great explorations there. I think I would probably change this wording a tiny bit:

> This color combination may be hard to read. Make it more accessible by using: 

To

> This color combination may be hard to read. Improve accessibility by using: 

It says the same thing and yet does it in a shorter phrasing. However, it would be great to get other copy reviews.

I am not sure about showing the colors, the reason for that is I think that they would have to read out to those using screenreaders and that could be a slightly strange experience here. I also think some colors are going to be tricky with the notice background.

**Comment by javierarce** [comment]
> This color combination may be hard to read. Improve accessibility by using: 
> It says the same thing and yet does it in a shorter phrasing. However, it would be great to get other copy reviews.

That's much better, thanks!

> I am not sure about showing the colors, the reason for that is I think that they would have to read out to those using screenreaders and that could be a slightly strange experience here. I also think some colors are going to be tricky with the notice background.

I think we could add the `aria-hidden="true"` attribute so they would be ignored by screen readers. And the border around the circles would help to make them distinguishable (we could tweak it a little bit to increase the contrast). 

In any case, we can go for a purely text-based message. The important thing is that the user responsible for selecting the color combination identifies the potential issue and can fix it easily.

**Comment by habarahonaa** [comment]
Is this issue still active? I would be happy to contribute. Thanks :)

**Comment by david-rahrer** [comment]
Is there any method to turn these warnings off? It keeps complaining about the colors but I've tested them and they are fine. I'm not changing them so It's really just annoying at this point. Is there a function somewhere that will shut them off? Thanks.
